src/app/extensions/quoting/shared/product/product-add-to-quote-dialog/product-add-to-quote-dialog.component.ts

```diff
@@ -48,11 +53,11 @@ export class ProductAddToQuoteDialogComponent implements OnInit, OnDestroy {
         this.patchForm(quote);
       });
 
-    this.activeQuoteRequest$
+    // make active quote request to selected
+    this.quotingFacade.activeQuoteRequest$
       .pipe(
-        filter(quoteRequest => !!quoteRequest),
-        distinctUntilKeyChanged('id'),
-        takeUntil(this.destroy$)
+        whenTruthy(),
+        take(1)
```

Comment:
`take(1)` instead of `takeUntil(this.destroy$)` solved the problem of not presenting the "Thank you for your quote" message if there is more than one unsubmitted quote request and instead switching to the next unsubmitted quote request.
But with that change the " Create New Request from Quote" in the dialog will create a new quote request but not switch to it in the dialog.

Reply:
The commit "removed unnecessarily used activeQuoteRequest$ in favour of selectedQuoteRequest$" should not have changed this behavior and was just introduced to clean up the disturbing usage of two different quote request observables.